components/camera/server_test.go

```diff
@@ -506,3 +506,17 @@ func TestServer(t *testing.T) {
 		test.That(t, err.Error(), test.ShouldContainSubstring, errStreamFailed.Error())
 	})
 }
+
+//nolint
+func TestCameraClientIsReadImager(t *testing.T) {
+	// RSDK-8663: Enforce that a camera client always satisfies the optimized `ReadImager`
+	// interface.
+	cameraClient, err := camera.NewClientFromConn(nil, nil, "", resource.Name{}, nil)
+	test.That(t, err, test.ShouldBeNil)
+
+	if _, isReadImager := cameraClient.(camera.ReadImager); isReadImager {
+		// Success
```

Comment:
How this code got this way is a bit of a long story -- it wasn't due to preference. I was trying to use the "test.ShouldImplement" assertion, that I think requires reflection and additional error handling. But ultimately failed at that. So the structure of this if-statement was already there and I just had to use a type-assertion instead and do nothing in the "then" clause.

That said, I do think writing conditionals in the affirmative (i.e: avoid double negatives) reads better. And in this case, a real writer is going to write the if-statement that way -- just with a non-empty "then".

But normally I wouldn't take that to the "extreme" of having an empty "then" clause. I just left it this way because I was fed up with writing an assertion.

Reply:
Haha the test lib use of reflection can be super funky

Got it